cmd/thanos/flags.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/improbable-eng/thanos/pkg/cluster"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+func registerServerFlags(cmd *kingpin.CmdClause) (grpcBindAddr *string, httpBindAddr *string, newPeerFn func(log.Logger, *prometheus.Registry, bool, string, bool) (*cluster.Peer, error)) {
```

Comment:
`registerDefaultServerFlags`? as these are not all the flags that are needed for the components.